Assets/Scripts/NodeFactory.cs

```diff
@@ -0,0 +1,14 @@
+﻿using UnityEngine;
+
+namespace Assets.Scripts
+{
+  public class NodeFactory : MonoBehaviour
```

Comment:
I don't like the name, this does not create Node objects. Consider renaming to NodeGameObjectFactory